flask/src/test/groovy/net/corda/gradle/flask/FlaskPluginTest.groovy

```diff
@@ -167,4 +174,32 @@ class FlaskPluginTest {
         }
         assertEquals("net.corda.gradle.flask.test.AlternativeMain", prop.mainClassName)
     }
+
+    @Test
+    @DisplayName("Check that shutdown hooks are executed on flask jars")
+    void shutdownHookTest() {
+        invokeGradle("shutdownHookTestJar")
+        String javaHome = System.getProperty("java.home")
+        String javaExecutable = javaHome + File.separator + "bin" + File.separator + "java"
+        Path jarFile = testProjectDir.resolve("build/libs/shutdownHookTest.jar")
+        Process process = new ProcessBuilder(Arrays.asList(javaExecutable, '-jar', jarFile.toString()))
+            .directory(testProjectDir.toFile())
+            .inheritIO()
+            .start()
+        Path testFile = testProjectDir.resolve("shutdown-hook-executed")
+        int totalWait = 0
+        final int timeoutMillis = 5000
+        while(!Files.exists(testFile)) {
+            int loopWait = 100
+            totalWait += loopWait
+            if(totalWait > timeoutMillis) {
+                throw new AssertionError("Failed to create test file '$testFile'")
+            }
+            Thread.sleep(loopWait)
+        }
+        process.destroy()
+        int returnCode = process.waitFor()
+        assertNotEquals(0, returnCode)
+        assertTrue(!Files.exists(testProjectDir.resolve("shutdown-hook-executed")))
```

Comment:
AssertJ has an `assertThat()` that can provide a better assertion for a `Path` than `assertTrue()`.